### PR TITLE
[Bugfix] Fixes For Changing Values On Number Input Field

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes fixes for clearing values in the number input field. Let me know your thoughts.

### CODE REVIEW

**Comment by Civolilah** [comment]
@turbo124 This PR also resolves this issue: https://invoiceninja.atlassian.net/browse/RU-1720.

File: src/components/forms/NumberInputField.tsx [hunk 1]
```diff
@@ -148,7 +128,35 @@ export function NumberInputField(props: Props) {
...
+                        symbol: '',
+                        precision:
+                          typeof props.precision === 'number'
+                            ? props.precision
+                            : reactSettings?.number_precision &&
+                              reactSettings?.number_precision > 0 &&
+                              reactSettings?.number_precision <= 100
+                            ? reactSettings.number_precision
+                            : 2,
+                      }).value
```
Comment (beganovich) [inline src/components/forms/NumberInputField.tsx:157]: This is really starting to get out of hand. Can we extract this into separate function with clean if statements?

**Review by beganovich — COMMENTED** [review]
Looks good, but we need tiny refactor.

File: src/components/forms/NumberInputField.tsx (hunk 1 above)
Comment (Civolilah) [reply src/components/forms/NumberInputField.tsx]: @beganovich Sure thing, the logic has been separated into a function and then applied to the logic. Let me know your thoughts.

**Comment by Civolilah** [comment]
Closes #2083

**Review by beganovich — APPROVED** [review]

**Pull request merged**